koku/api/provider/models.py

```diff
@@ -84,16 +85,23 @@ def get_queryset(self):
         """Return a Queryset of non-OCP and active and non-paused Providers."""
         return super().get_queryset().filter(active=True, paused=False).exclude(type=Provider.PROVIDER_OCP)
 
-    def get_polling_batch(self, limit=-1, offset=0, filters=None):
+    def get_polling_batch(self, filters=None):
         """Return a Queryset of pollable Providers that have not polled in the last 24 hours."""
         polling_delta = datetime.now(tz=settings.UTC) - timedelta(seconds=settings.POLLING_TIMER)
         if not filters:
             filters = {}
-        if limit < 1:
+        # Dynamically set batch limit (divide count by 21 hours so we always trigger a few extra + round up)
+        batch_limit = math.ceil(len(self.filter(**filters)) / 21)
+        if batch_limit <= 1:
             # Django can't do negative indexing, so just return all the Providers.
             # A limit of 0 doesn't make sense either. That would just return an empty QuerySet.
```

Comment:
If you're changing the functionality of this function, then this comment should be removed. It no longer applies here.

Reply:
Sure thing.